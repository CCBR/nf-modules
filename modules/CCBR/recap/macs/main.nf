process RECAP_MACS {

    container 'nciccbr/recap:0.3.0'

    input:
        tuple val(meta), path(chip), path(input)

    output:
        tuple val(meta), path("${meta.id}/MACS_RECAP/*.xls"),                                        emit: recap
        tuple val(meta), path("${meta.id}/MACS_RECAP/${chip.baseName}.RECAP.bootstrap_1_peaks.bed"), emit: peaks

    script:
    def outdir = "${meta.id}"
    def recap_out = "${outdir}/MACS_RECAP/${chip.baseName}.RECAP.bootstrap_1_peaks.xls"
    def recap_bed = "${outdir}/MACS_RECAP/${chip.baseName}.RECAP.bootstrap_1_peaks.bed"
    """
    mkdir -p ${outdir}
    RECAP_MACS.sh \\
        -i `realpath ./` \\
        -t ${chip} \\
        -c ${input} \\
        -o `realpath ${outdir}` \\
        -b 1 -e 29 \\
        -s 20241211
    recap_to_bed.py ${recap_out} ${recap_bed}
    """
}
