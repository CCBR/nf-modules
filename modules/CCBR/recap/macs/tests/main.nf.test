// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test recap/macs
nextflow_process {

    name "Test Process RECAP_MACS"
    script "../main.nf"
    process "RECAP_MACS"

    tag "modules"
    tag "modules_CCBR"
    tag "recap"
    tag "recap/macs"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("hsapiens") {

        // TODO nf-core: If you are created a test for a chained module
        // (the module requires running more than one process to generate the required output)
        // add the 'setup' method here.
        // You can find more information about how to use a 'setup' method in the docs (https://nf-co.re/docs/contributing/modules#steps-for-creating-nf-test-for-chained-modules).

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.test_data_base + '/data/genomics/homo_sapiens/pacbio/bed/alz.ccs.fl.NEB_5p--NEB_Clontech_3p.flnc.clustered.singletons.merged.aligned_tc.bed', checkIfExists: true) ],
                    [ file(params.test_data_base + '/data/genomics/homo_sapiens/pacbio/bed/alz.ccs.fl.NEB_5p--NEB_Clontech_3p.flnc.clustered.singletons.merged.aligned_tc.2.bed', checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("hsapiens stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.test_data_base + '/data/genomics/homo_sapiens/pacbio/bed/alz.ccs.fl.NEB_5p--NEB_Clontech_3p.flnc.clustered.singletons.merged.aligned_tc.bed', checkIfExists: true) ],
                    [ file(params.test_data_base + '/data/genomics/homo_sapiens/pacbio/bed/alz.ccs.fl.NEB_5p--NEB_Clontech_3p.flnc.clustered.singletons.merged.aligned_tc.2.bed', checkIfExists: true) ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
