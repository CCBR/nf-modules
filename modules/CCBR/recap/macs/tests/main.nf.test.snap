{
    "hsapiens": {
        "content": [
            {
                "0": [

                ],
                "1": [

                ],
                "peaks": [

                ],
                "recap": [

                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-11T16:34:47.684081028"
    },
    "hsapiens stub": {
        "content": [
            {
                "0": [

                ],
                "1": [

                ],
                "peaks": [

                ],
                "recap": [

                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-12-11T16:34:53.055465915"
    }
}
